src/components/Extensibility/components-form/Modules/Modules.js

```diff
@@ -0,0 +1,226 @@
+import React from 'react';
+import { useUIStore } from '@ui-schema/ui-schema';
+
+import { useJsonata } from '../../hooks/useJsonata';
+import { useVariables } from '../../hooks/useVariables';
+import { fromJS } from 'immutable';
+import {
+  ComboboxInput,
+  MessageStrip,
+  Checkbox,
+  Link,
+  Icon,
+} from 'fundamental-react';
+
+import './Modules.scss';
+import { useGetTranslation } from 'components/Extensibility/helpers';
+import { ResourceForm } from 'shared/ResourceForm';
+
+export function Modules({
+  storeKeys,
+  resource,
+  onChange,
+  schema,
+  required,
+  editMode,
+}) {
+  const { t: tExt } = useGetTranslation();
+  const sectionName = schema.get('name');
+
+  const setCheckbox = (fullValue, key, entryValue, checked, index) => {
+    if (checked) {
+      onChange({
+        storeKeys,
+        scopes: ['value', 'internal'],
+        type: 'list-item-add',
+        schema,
+        itemValue: fromJS({ [key]: entryValue }),
+        required,
+      });
+    } else {
+      onChange({
+        storeKeys,
+        scopes: ['value', 'internal'],
+        type: 'list-item-delete',
+        index,
+        schema,
+        required,
+      });
+    }
+  };
+  const { store } = useUIStore();
+  const { value } = store?.extractValues(storeKeys) || [];
+  const { itemVars } = useVariables();
+  const jsonata = useJsonata({
+    resource,
+    scope: value,
+    value,
+  });
+
+  const rule = schema.get('schemaRule');
+
+  const options = schema.get('options');
+  let parsedOptions = {};
+
+  function makeJsonata(propObject) {
+    if (typeof propObject === 'string') {
+      const [newEnum] = jsonata(
+        propObject,
+        itemVars(resource, rule.itemVars, storeKeys),
+        [],
+      );
+      return newEnum;
+    }
+    console.warn('Widget::Modules');
+    return null;
+  }
+
+  Object.keys(options).forEach(optionName => {
+    parsedOptions[optionName] = makeJsonata(options[optionName]);
+  });
+
+  const Items = parsedOptions?.name?.map((name, index) => {
+    const isChecked = !!(value ? value.toJS() : []).find(v => {
+      return v.name === name;
+    });
+
+    let channelTest = [];
+
+    parsedOptions?.moduleTemplates?.map(moduleTemplate => {
+      if (
+        moduleTemplate?.metadata?.labels[
+          'operator.kyma-project.io/module-name'
+        ] === name
+      ) {
+        if (moduleTemplate?.spec?.descriptor?.component?.version)
+          return channelTest.push({
+            text: moduleTemplate.spec.channel,
+            additionalText: `(version: ${moduleTemplate?.spec?.descriptor?.component?.version})`,
+          });
+
+        return channelTest.push({
+          text: moduleTemplate.spec.channel,
+          additionalText: '',
+        });
+      } else {
+        return '';
+      }
+    });
```

Comment:
Could be simplified to a filter and map
```suggestion
    const channelTest = parsedOptions?.moduleTemplates?.filter(moduleTemplate => moduleTemplate?.metadata?.labels['operator.kyma-project.io/module-name'] === name).map(moduleTemplate => {
        const additionalText = moduleTemplate?.spec?.descriptor?.component?.version?
            `(version: ${moduleTemplate?.spec?.descriptor?.component?.version})` :
            '';
        return channelTest.push({
          text: moduleTemplate.spec.channel,
          additionalText,
        });
    });
```